backend/de.metas.business/src/main/java/de/metas/pricing/rules/campaign_price/impexp/CampaignPriceImportTableSqlUpdater.java

```diff
@@ -0,0 +1,222 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.pricing.rules.campaign_price.impexp;
+
+import de.metas.impexp.processing.ImportRecordsSelection;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.experimental.UtilityClass;
+import org.adempiere.ad.trx.api.ITrx;
+import org.compiere.model.I_I_Campaign_Price;
+import org.compiere.util.DB;
+
+import java.util.Properties;
+
+import static de.metas.impexp.format.ImportTableDescriptor.COLUMNNAME_I_ErrorMsg;
+import static de.metas.impexp.format.ImportTableDescriptor.COLUMNNAME_I_IsImported;
+
+@UtilityClass
+public class CampaignPriceImportTableSqlUpdater
+{
+	private final static String targetTableName = I_I_Campaign_Price.Table_Name;
+
+	@Builder(buildMethodName = "updateCampaignPrice")
+	private void updateCampaignPrice(@NonNull final ImportRecordsSelection selection, @NonNull final Properties ctx)
+	{
+		dbUpdateProduct(selection);
+		dbUpdateBPartner(selection);
+		dbUpdateBpGroup(selection);
+		dbUpdatePricingSystem(selection);
+		dbUpdateCountry(selection);
+		dbUpdateCurrency(selection);
+		dbUpdateUom(selection);
+		dbUpdateTaxCategory(selection);
+		dbUpdateCampaignPrice(selection, ctx);
+	}
+
+	private static void dbUpdateProduct(final ImportRecordsSelection selection)
+	{
+		StringBuilder sql = new StringBuilder("UPDATE ")
+				.append(targetTableName + " i ")
+				.append(" SET M_Product_ID=(SELECT M_Product_ID FROM M_Product p")
+				.append(" WHERE i.ProductValue=p.Value AND p.AD_Client_ID in (i.AD_Client_ID,0) AND p.AD_Org_ID IN (0, i.AD_Org_ID) LIMIT 1) ")
+				.append("WHERE M_Product_ID IS NULL")
+				.append(" AND " + COLUMNNAME_I_IsImported + "='N'")
+				.append(selection.toSqlWhereClause("i"));
+		DB.executeUpdateEx(sql.toString(), ITrx.TRXNAME_ThreadInherited);
+
+		sql = new StringBuilder("UPDATE ")
+				.append(targetTableName)
+				.append(" SET " + COLUMNNAME_I_IsImported + "='E', " + COLUMNNAME_I_ErrorMsg + "=" + COLUMNNAME_I_ErrorMsg + "||'ERR=Invalid Product,' ")
+				.append("WHERE M_Product_ID IS NULL AND ProductValue IS NOT NULL")
+				.append(" AND " + COLUMNNAME_I_IsImported + "<>'Y'")
+				.append(selection.toSqlWhereClause());
+		DB.executeUpdateEx(sql.toString(), ITrx.TRXNAME_ThreadInherited);
+	}
+
+	private static void dbUpdateBPartner(final ImportRecordsSelection selection)
+	{
+		StringBuilder sql = new StringBuilder("UPDATE ")
+				.append(targetTableName + " i ")
+				.append(" SET C_BPartner_ID=(SELECT C_BPartner_ID FROM C_BPartner p")
+				.append(" WHERE i.BPartner_Value=p.Value AND p.AD_Client_ID in (i.AD_Client_ID, 0) AND p.AD_Org_ID IN (0, i.AD_Org_ID) LIMIT 1) ")
```

Comment:
see the comment about `dbUpdateProduct()`

Reply:
Done
![image](https://user-images.githubusercontent.com/60300682/165274853-1c594921-2752-4743-8f3b-19f71bf73f8e.png)